common/txmgr/broadcaster.go

```diff
@@ -287,30 +285,37 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) Trig
 }
 
 // Load the next sequence map using the tx table or on-chain (if not found in tx table)
-func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) loadNextSequenceMap(addresses []ADDR) (map[ADDR]SEQ, error) {
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) loadNextSequenceMap(addresses []ADDR) map[ADDR]SEQ {
 	ctx, cancel := eb.chStop.NewCtx()
 	defer cancel()
 
 	nextSequenceMap := make(map[ADDR]SEQ)
 	for _, address := range addresses {
-		// Get the highest sequence from the tx table
-		// Will need to be incremented since this sequence is already used
-		seq, err := eb.txStore.FindLatestSequence(ctx, address, eb.chainID)
-		if err != nil {
-			// Look for nonce on-chain if no tx found for address in TxStore or if error occurred
-			// Returns the nonce that should be used for the next transaction so no need to increment
-			seq, err = eb.client.PendingSequenceAt(ctx, address)
-			if err != nil {
-				return nil, errors.New("failed to retrieve next sequence from on-chain causing failure to load next sequence map on broadcaster startup")
-			}
-
-			nextSequenceMap[address] = seq
-		} else {
-			nextSequenceMap[address] = eb.generateNextSequence(seq)
+		seq := eb.getSequenceForAddr(ctx, address)
+		if seq != nil {
+			nextSequenceMap[address] = *seq
 		}
 	}
 
-	return nextSequenceMap, nil
+	return nextSequenceMap
+}
+
+func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) getSequenceForAddr(ctx context.Context, address ADDR) *SEQ {
+	// Get the highest sequence from the tx table
+	// Will need to be incremented since this sequence is already used
+	seq, err := eb.txStore.FindLatestSequence(ctx, address, eb.chainID)
+	if err != nil {
+		// Look for nonce on-chain if no tx found for address in TxStore or if error occurred
+		// Returns the nonce that should be used for the next transaction so no need to increment
+		seq, err = eb.client.PendingSequenceAt(ctx, address)
+		if err != nil {
+			eb.logger.Criticalw("failed to retrieve next sequence from on-chain for address: ", "address", address.String())
+			return nil
+		}
+		return &seq
+	}
+	seq = eb.generateNextSequence(seq)
+	return &seq
```

Comment:
IMHO "return early, return often" makes this more readable:
```suggestion
	if err == nil {
		seq = eb.generateNextSequence(seq)
		return &seq
	}
	// Look for nonce on-chain if no tx found for address in TxStore or if error occurred
	// Returns the nonce that should be used for the next transaction so no need to increment
	seq, err = eb.client.PendingSequenceAt(ctx, address)
	if err != nil {
		eb.logger.Criticalw("failed to retrieve next sequence from on-chain for address: ", "address", address.String())
		return nil
	}
	return &seq
```

Reply:
Sorry I should have addressed this with your comment for `GetNextSequence` but I've updated this too now